libs/liquidity/src/lib/liquidity-data-provider.ts

```diff
@@ -0,0 +1,231 @@
+import { gql } from '@apollo/client';
+import {
+  addDecimalsFormatNumber,
+  formatNumberPercentage,
+} from '@vegaprotocol/react-helpers';
+import type { LiquidityProvisionStatus } from '@vegaprotocol/types';
+import { LiquidityProvisionStatusMapping } from '@vegaprotocol/types';
+import { AccountType } from '@vegaprotocol/types';
+import { useNetworkParameter } from '@vegaprotocol/web3';
+import BigNumber from 'bignumber.js';
+import { SISKA_NETWORK_PARAMETER } from './liquidity-manager';
+import type {
+  MarketLiquidity,
+  MarketLiquidity_market_data_liquidityProviderFeeShare,
+} from './__generated__';
+
+export const MARKET_LIQUIDITY_QUERY = gql`
+  query MarketLiquidity($marketId: ID!, $partyId: String) {
+    market(id: $marketId) {
+      id
+      decimalPlaces
+      positionDecimalPlaces
+      liquidityProvisionsConnection(party: $partyId) {
+        edges {
+          node {
+            id
+            party {
+              id
+              accountsConnection(marketId: $marketId, type: ACCOUNT_TYPE_BOND) {
+                edges {
+                  node {
+                    type
+                    balance
+                  }
+                }
+              }
+            }
+            createdAt
+            updatedAt
+            commitmentAmount
+            fee
+            status
+          }
+        }
+      }
+      tradableInstrument {
+        instrument {
+          code
+          product {
+            ... on Future {
+              settlementAsset {
+                id
+                symbol
+                decimals
+              }
+            }
+          }
+        }
+      }
+      data {
+        market {
+          id
+        }
+        suppliedStake
+        openInterest
+        targetStake
+        marketValueProxy
+        liquidityProviderFeeShare {
+          party {
+            id
+          }
+          equityLikeShare
+          averageEntryValuation
+        }
+      }
+    }
+  }
+`;
+
+export interface LiquidityProvision {
+  party: string;
+  commitmentAmount: string | undefined;
+  fee: string | undefined;
+  equityLikeShare: string;
+  averageEntryValuation: string;
+  obligation: string | null;
+  supplied: string | null;
+  status: LiquidityProvisionStatus | string | undefined;
```

Comment:
question (blocking): Why type this as also string | undefined? The api indicates that this can only every be `LiquidityProvisionStatus`. Some thing is wrong in the types elsewhere if you've had to do this. If its because node can be undefined then you should filter those nodes out first.

Reply:
for every data.liquidityProviderFeeShare we are trying to find the matching liquidity provision connection node. the API does not match it for us. I tried to filter non-empty connection nodes, however typescript being typescript still complains if I leave the connection as not being possibly null/undefined.